### PR TITLE
Custom version v2

### PR DESCRIPTION
**Pull request by PSRCode**

### CODE REVIEW

File: include/Makefile.am [hunk 1]
```diff
@@ -30,7 +30,7 @@
 ## Data fetched from "version/extra_version_description" must be sanitized and
 ## formatted.
 ## The sanitize step uses sed with a script to replace all
-## non-alpha-numeric values to "-".
+## non-alpha-numeric, execpt " " (space), values to "-".
```
Comment (mjeanson) [inline include/Makefile.am]: execpt -> except

File: include/Makefile.am [hunk 2]
```diff
@@ -34,6 +34,15 @@
 ## The formatting step consist of using sed with a script that join all lines
 ## by replacing "\n" with litteral string c-style "\n\t".
 ##
+## Repeat the same logic for the "version/extra_modifications" directory.
+## Data fetched from "version/extra_modifications" must be sanitized and
+## formatted.
+## The data is fectched using "ls" with an ignore pattern for the README file.
+## The sanitize step uses ised with a script to replace all
```
Comment (mjeanson) [inline include/Makefile.am]: ised?

File: include/Makefile.am [hunk 3]
```diff
@@ -34,6 +34,15 @@
 ## The formatting step consist of using sed with a script that join all lines
 ## by replacing "\n" with litteral string c-style "\n\t".
 ##
+## Repeat the same logic for the "version/extra_modifications" directory.
+## Data fetched from "version/extra_modifications" must be sanitized and
+## formatted.
+## The data is fectched using "ls" with an ignore pattern for the README file.
```
Comment (mjeanson) [inline include/Makefile.am]: fectched -> fetched

File: include/Makefile.am [hunk 4]
```diff
@@ -20,11 +20,24 @@
...
 ##
 ## If we don't have a "version.i.tmp" nor a "version.i", generate an empty
-## string as a failover.
+## string as a failover. If a "version.i" is present, for example when building
+## from a distribution tarball, get the git_version using grep.
 ##
-## If we don't have a "version.i" or we have both files and they are different,
-## copy "version.i.tmp" over "version.i". This way the dependent targets are
-## only rebuilt when the version string changes.
+## Fetch to value for the EXTRA_VERSION_NAME define from
```
Comment (mjeanson) [inline include/Makefile.am]: Fetch the value?

File: configure.ac [hunk 5]
```diff
@@ -1002,7 +1002,13 @@ AM_CONDITIONAL([BUILD_LIB_UST_CONSUMER], [test x$build_lib_ust_consumer = xyes])
 AM_CFLAGS="-Wall -fno-strict-aliasing $PTHREAD_CFLAGS"
 AC_SUBST(AM_CFLAGS)
 
-AM_CPPFLAGS="-I\$(top_srcdir)/include -I\$(top_builddir)/include -I\$(top_srcdir)/src -include config.h $AM_CPPFLAGS"
+# The order in which the include folders are searched is important.
+# The top_builddir should always be searched first in the event that a generated
+# file at build time is included. An example of this is the "version.i" file. In a
```
Comment (mjeanson) [inline configure.ac]: a build time generated file is included

File: configure.ac [hunk 6]
```diff
@@ -1002,7 +1002,13 @@ AM_CONDITIONAL([BUILD_LIB_UST_CONSUMER], [test x$build_lib_ust_consumer = xyes])
...
+# scenario where lttng-tools is built from a distribution tarball and in a
+# out-of-tree manner, the generated "version.i" have priority on the one from
```
Comment (mjeanson) [inline configure.ac]: has priority

Comment (PSRCode) [reply configure.ac]: done

File: configure.ac (hunk 5 above)
Comment (PSRCode) [reply configure.ac]: done

File: include/Makefile.am (hunk 4 above)
Comment (PSRCode) [reply include/Makefile.am]: done

File: include/Makefile.am (hunk 2 above)
Comment (PSRCode) [reply include/Makefile.am]: done

File: include/Makefile.am (hunk 1 above)
Comment (PSRCode) [reply include/Makefile.am]: done

File: include/Makefile.am (hunk 3 above)
Comment (PSRCode) [reply include/Makefile.am]: done

**Comment by PSRCode** [comment]
Mjeanson will open a new v3

**Pull request closed without merging**